components/CartPopupModal/CartPopupModal.js

```diff
@@ -151,6 +246,67 @@ const CartPopupModal = ({ type, item, selector, setCartState, disabled }) => {
         {type}
       </Button>
 
+      <Dialog
+        open={openConfirmation}
+        onClose={handleCloseConfirmation}
+        aria-labelledby='responsive-dialog-title'
+      >
+        <DialogTitle id='responsive-dialog-title'>
+          {'Proceed to deposit item?'}
+        </DialogTitle>
+        <DialogContent>
+          <DialogContentText>
+            For new expiry date, the deposit will take effect immediately once
+            you submit and will hence not be added to your cart. Do you still
+            wish to submit?
+          </DialogContentText>
+        </DialogContent>
+        <DialogActions>
+          <Button autoFocus onClick={handleCloseConfirmation}>
+            Cancel
+          </Button>
+          <Button
+            onClick={() => {
+              formik.handleSubmit();
+              handleCloseConfirmation();
+            }}
+            autoFocus
+          >
+            Submit
+          </Button>
+        </DialogActions>
+      </Dialog>
+
+      <Dialog
+        open={openDatePicker}
+        onClose={() => handleCloseDatePicker('close')}
+        aria-labelledby='responsive-dialog-title'
+      >
+        <DialogTitle id='responsive-dialog-title'>
+          {'Pick a new expiry date'}
+        </DialogTitle>
+        <DialogContent>
+          <LocalizationProvider dateAdapter={AdapterDayjs}>
```

Comment:
think it would be better to put this in the top level, ie in `pages/inventory/itemIndex.js`